File: src/test/kotlin/fr/unilim/saes5/unit/WordAnalyticsTest.java
package fr.unilim.saes5.unit;

import fr.unilim.saes5.interfaces.IAnalyzable;
import fr.unilim.saes5.model.Word;
import fr.unilim.saes5.model.WordAnalytics;
import fr.unilim.saes5.model.reader.TestReader;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.HashMap;
import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

public class WordAnalyticsTest {

    private static WordAnalytics wordAnalytics;

    @BeforeEach
    void init() {
        WordAnalyticsTest.wordAnalytics = new WordAnalytics();
    }


    @Test
    void testWordPresent() {
        List<Word> words = new ArrayList<Word>();
        words.add(new Word("Jeu"));
        words.add(new Word("Plateau"));
        words.add(new Word("Grille"));

        assertThat(WordAnalyticsTest.wordAnalytics.isWordPresent(words)).isTrue();
    }

    @Test
    void testWordRatio() {
        IAnalyzable reader = new TestReader(Arrays.asList(new Word("Robot"), new Word("Joueur")));
        assertThat(WordAnalyticsTest.wordAnalytics.wordRatio(new Word("Robot"), reader.read())).isEqualTo(50.00f); // tester avec une liste de mot qui contient juste Robot
    }

    @Test
    void testWordRank() {
        IAnalyzable reader = new TestReader(Arrays.asList(new Word("Robot"), new Word("Robot"), new Word("Joueur")));
        HashMap<Word, Integer> expected = new HashMap<Word, Integer>();
        expected.put(new Word("Robot"), 2);
        expected.put(new Word("Joueur"), 1);

        assertThat(WordAnalyticsTest.wordAnalytics.wordRank(reader.read())).isEqualTo(expected);
    }
}

File: src/test/kotlin/fr/unilim/saes5/unit/WordAnalyticsTest.kt
package fr.unilim.saes5.unit

import fr.unilim.saes5.model.interfaces.IRead
import fr.unilim.saes5.model.Word
import fr.unilim.saes5.model.WordAnalytics
import fr.unilim.saes5.model.reader.DummyReader
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*

class WordAnalyticsTest {
    @BeforeEach
    fun init() {
        wordAnalytics = WordAnalytics()
    }

    @Test
    fun testWordPresent() {
        val words: MutableList<Word?> = ArrayList()
        words.add(Word("Jeu"))
        words.add(Word("Plateau"))
        words.add(Word("Grille"))
        for (word in words) {
            Assertions.assertThat(wordAnalytics!!.isWordPresent(words, word)).isTrue()
        }
    }

    @Test
    fun testWordRatio() {
        val reader: IRead = DummyReader(listOf(Word("Robot"), Word("Joueur")))
        Assertions.assertThat(wordAnalytics!!.wordRatio(Word("Robot"), reader.read("")))
            .isEqualTo(50.00f) // tester avec une liste de mot qui contient juste Robot
    }

    @Test
    fun testWordRank() {
        val reader: IRead = DummyReader(Arrays.asList(Word("Robot"), Word("Robot"), Word("Joueur")))
        val expected = HashMap<Word, Int>()
        expected[Word("Robot")] = 2
        expected[Word("Joueur")] = 1
        Assertions.assertThat(wordAnalytics!!.wordRank(reader.read(""))).isEqualTo(expected)
    }

    companion object {
        private var wordAnalytics: WordAnalytics? = null
    }
}
